iroha_http_server/src/lib.rs

```diff
@@ -134,28 +138,65 @@ pub mod http {
             path_params: PathParams,
             query_params: QueryParams,
             request: HttpRequest,
-        ) -> Result<HttpResponse>;
+        ) -> HttpResponse;
     }
 
     #[async_trait]
     impl<State, F, Fut> HttpHandler<State> for F
     where
         State: Clone + Send + Sync + 'static,
         F: Send + Sync + 'static + Fn(State, PathParams, QueryParams, HttpRequest) -> Fut,
-        Fut: Future<Output = Result<HttpResponse>> + Send + 'static,
+        Fut: Future<Output = HttpResponse> + Send + 'static,
     {
         async fn call(
             &self,
             state: State,
             path_params: PathParams,
             query_params: QueryParams,
             request: HttpRequest,
-        ) -> Result<HttpResponse> {
+        ) -> HttpResponse {
             let future = (self)(state, path_params, query_params, request);
             future.await
         }
     }
 
+    pub trait HttpResponseError: Send + Sync + 'static {
+        fn status_code(&self) -> StatusCode {
+            HTTP_CODE_INTERNAL_SERVER_ERROR
+        }
+
+        fn reason(code: StatusCode) -> String {
+            match code {
+                HTTP_CODE_INTERNAL_SERVER_ERROR => "Internal server error",
+                HTTP_CODE_NOT_FOUND => "Not found",
+                HTTP_CODE_BAD_REQUEST => "Bad request",
+                HTTP_CODE_METHOD_NOT_ALLOWED => "Method not allowed",
+                HTTP_CODE_UPGRADE_REQUIRED => "Upgrade required",
+                _ => unimplemented!(),
+            }
+            .to_owned()
+        }
+
+        fn error_body(&self) -> Vec<u8>;
+
+        fn error_response(&self) -> HttpResponse {
+            let code = self.status_code();
+            HttpResponse {
+                version: HttpVersion::Http1_1,
+                code,
+                reason: Self::reason(code),
+                headers: Headers::new(),
+                body: self.error_body(),
+            }
+        }
+    }
+
+    impl HttpResponseError for std::convert::Infallible {
```

Comment:
Please correct `std::convert::Infallible` according to the style that I mentioned previously. Also check if any other places like this are left.